ls1intum/Artemis
# `Iris`: Add clear chat session button

### Checklist
#### General


#### Server
#### Client
#### Changes affecting Programming Exercises

### Motivation and Context


Add clear chat history button. In the background, it will call API from the server to create a new session

### Description

- Add a new button and call API on click

### Steps for Testing

Prerequisites:
- 1 Student
- 1 Programming Exercise with Iris enabled

1. Log in to Artemis
2. Navigate to a programming exercise with Iris enabled
3. Chat something and then click on the Trash button
4. Click submit on the confirm modal
5. Your chat history should disappear and you can chat with the new context
6. The trash button is disappear
7. Refresh the page, accept the policy, and even with Iris welcome message the trash button still disappears

It should be able to work on all sizes of the chatbot

#### Exam Mode Testing



### Review Progress






#### Performance Review
#### Code Review
#### Manual Tests
#### Exam Mode Test

### Test Coverage







### Screenshots


![image](https://github.com/ls1intum/Artemis/assets/18320285/e29e7183-69d5-4fb1-ad13-f291c6890bd5)

![image](https://github.com/ls1intum/Artemis/assets/18320285/e630833d-6f5a-4ae3-9682-c4e94d4d6723)
